src/Core/Compiler/CompilerService.cs

```diff
@@ -141,15 +141,16 @@ public IEnumerable<QsNamespaceElement> IdentifyElements(string source)
                 LoadReferencesBasedOnGeneratedCsharp = false, // deserialization of resources in references is only needed if there is an execution target
                 IsExecutable = compileAsExecutable,
                 AssemblyConstants = new Dictionary<string, string> { [AssemblyConstants.ProcessorArchitecture] = executionTarget ?? string.Empty },
-                RuntimeCapability = runtimeCapability ?? RuntimeCapability.FullComputation
+                RuntimeCapability = runtimeCapability ?? RuntimeCapability.FullComputation,
+                PrepareQirGeneration = executionTarget?.StartsWith("microsoft.simulator") ?? false
```

Comment:
This should be abstracted in a `TargetSupportsQir` method instead of having a hardcoded value. That way we can just update that method whenever a new QIR target becomes available.

Reply:
I see the hardcoded value you are referring to here, but I don't understand your suggestion regarding a `TargetSupportsQir` method. What object would the method live on, and what would the method do?

Reply:
The ideal place for this method to live would be here: [SubmitterFactory.cs](https://github.com/microsoft/qsharp-runtime/blob/main/src/Azure/Azure.Quantum.Client/SubmitterFactory.cs). It is a cross-repo change so I suggest adding a ToDo comment here and create the corresponding issue.

Reply:
The logic for determining if we should generate QIR should live in the SubmitterFactory as a static function, `TargetSupportsQir`. We should then be calling that static function here for PrepareQirGeneration instead of this logic.